ios/RNTrackPlayer/RNTrackPlayer.swift

```diff
@@ -185,159 +187,184 @@ public class RNTrackPlayer: RCTEventEmitter, MediaWrapperDelegate {
         }
         
         print("Adding tracks:", tracks)
-        mediaWrapper.addTracks(tracks, before: trackId)
+        
+        if let trackId = trackId {
+            guard let insertIndex = player.queueManager.items.firstIndex(where: { ($0 as! Track).id == trackId })
+            else {
+                reject("track_not_in_queue", "Given track ID was not found in queue", nil)
+                return
+            }
+            
+            try? player.add(items: tracks, at: insertIndex)
+        } else {
+            try? player.add(items: tracks)
+        }
+        
         resolve(NSNull())
     }
     
     @objc(remove:resolver:rejecter:)
     public func remove(tracks ids: [String], resolve: RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) {
         print("Removing tracks:", ids)
-        mediaWrapper.removeTracks(ids: ids)
+        var indexesToRemove: [Int] = []
+        
+        for id in ids {
+            if let index = player.queueManager.items.firstIndex(where: { ($0 as! Track).id == id }) {
+                if index == player.queueManager.currentIndex { return }
```

Comment:
Is it not rejecting the promise? Which code should we use here?

Reply:
I think we should not reject. We should just ignore `ids` that we don't find and if the `id` corresponds to the current track